kover-gradle-plugin/src/functionalTest/kotlin/kotlinx/kover/gradle/plugin/test/functional/cases/VerificationTests.kt

```diff
@@ -175,6 +175,71 @@ Rule violated: lines missed count for package 'org.jetbrains.kover.test.function
         }
     }
 
+    @SlicedGeneratedTest(allLanguages = true, allTools = true)
+    fun BuildConfigurator.testRootRulesWarnOnFail() {
+        addProjectWithKover {
+            sourcesFrom("simple")
+
+            kover {
+                reports {
+                    verify {
+                        warnOnFailure.set(true)
+                        rule("root rule") {
+                            bound {
+                                minValue.set(99)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        run("koverVerify") {
+            taskOutput(":koverVerify") {
+                match {
```

Comment:
Should we also check that build finished successfully?

Reply:
Test will fail if build finished unsuccessfully